easybuild/framework/easyconfig/tweak.py

```diff
@@ -94,18 +94,15 @@ def tweak(easyconfigs, build_specs, modtool, targetdir=None):
     # keep track of originally listed easyconfigs (via their path)
     listed_ec_paths = [ec['spec'] for ec in easyconfigs]
 
-    # obtain full dependency graph for specified easyconfigs
-    # easyconfigs will be ordered 'top-to-bottom': toolchain dependencies and toolchain first
-    orig_ecs = resolve_dependencies(easyconfigs, modtool, retain_all_deps=True)
```

Comment:
this is actually a bug, this line is overwriting the `orig_ecs` defined above in the `if-else`...